### ISSUE TITLE
Proposals can be cloned and submitted with instruments that aren't on the call

### ISSUE THREAD

**Issue by ACLay** [UserOfficeProject/user-office-project-issue-tracker#1192]
### What is the problem and why is it a problem

For ISIS direct 2025_1 we excluded the LET instrument. An old proposal that used that instrument was cloned to create 2510291, and it was able to submit with that invalid instrument.

I think @TCMeldrum can fill in this, but there seem to be some broken links as a result of this, causing the proposal to be unqueriable in our downstream software?

**Comment by TCMeldrum** [comment]
After some investigation of what happened here is what I think happened, some of this is STFC specific so anyone else reading this I'm sorry. 

- A user cloned a proposal from a old call where they had selected a instrument `Let` which was not being offered on this call
- The system allowed this proposal to be submitted **(This is the bug)** 
- The automatic instrument assignment didn't happen as the instrument is not on the call so this proposal ended up submitted but with no instrument assigned to it
 
STFC Specific: 

- Allocations then tried processing the proposal but since [no instrument was assigned to the proposal](https://github.com/isisbusapps/proposal-allocations/blob/c8e0652e1494297357cd6751a025b0285689d501/proposal-allocations/src/main/java/uk/ac/stfc/facilities/allocations/proposalsintergration/ProposalSubmissionEventHandler.java#L243), allocations defaulted on looking for an [instrument selection question](https://github.com/isisbusapps/proposal-allocations/blob/c8e0652e1494297357cd6751a025b0285689d501/proposal-allocations/src/main/java/uk/ac/stfc/facilities/allocations/proposalsintergration/ProposalSubmissionEventHandler.java#L196)
- This then failed as the call does not use these questions any more, so gave us an error saying template malformed and required answer not available. 

In conclusion, the bug in UOP was that a proposal can be submitted with an instrument picker answer that is invalid. 

For the allocations side this seems like the behaviour is not quiet right but idk what to do about it @ACLay

**Comment by mutambaraf** [comment]
To check with @Cat-Lucifer by @mehta-pooja123

**Comment by mutambaraf** [comment]
We have added validation of available instruments on proposals when saving .This  will require users to select available instruments when submitting.
